### PR TITLE
fix(igxHierarchicalGrid): Fix navigation with tab is not correct #4780

### LINKED ISSUE #4780: Navigation with Tab is not correct when a row is expanded
## Description  
Navigation with Tab is not correct when a row is expanded

 * igniteui-angular version: 7.3
 * browser: n/a

## Steps to reproduce  

1.Open dev sample hierarchicalGridUpdating
2. Expand row with id AROUT
3. In the row with BERGS click on cell in column Phone 
4. Press Tab key

## Result  
Cell in the child grid is selected

## Expected result  
Cell in row with id BLAUS and column CustemerId should be selected

### PR DESCRIPTION
**Pull request by mddragnev**
Closes #4780 

### Additional information (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

**Comment by skrustev** [comment]
This should already be fixed by PR #5900. The current fix is not taken into account because navigating using tab does not go through the igxHierarchicalGrid's navigation service, but through the regular igxGrid navigation service which means overriding the `getNextRowByIndex` won't affect the behavior.

File: projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.navigation.spec.ts [hunk 1]
```diff
@@ -1289,3 +1335,31 @@ export class IgxHierarchicalGridMultiLayoutComponent extends IgxHierarchicalGrid
     </igx-hierarchical-grid>`
 })
 export class IgxHierarchicalGridSmallerChildComponent extends IgxHierarchicalGridTestBaseComponent {}
+
+@Component({
```
Comment (nrobakova) [inline projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.navigation.spec.ts]: Can you use IgxHierarchicalGridTestComplexComponent istead of creating new one

File: projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.navigation.spec.ts [hunk 2]
```diff
@@ -1192,6 +1192,52 @@ describe('IgxHierarchicalGrid Smaller Child Navigation #hGrid', () => {
...
+    }));
+
+    it('should navigate to the first cell of next row using Tab from last cell in the row above', (async () => {
+        const row = hierarchicalGrid.getRowByIndex(0);
+        (row as IgxHierarchicalRowComponent).toggle();
+        await wait(100);
+        fixture.detectChanges();
+
+        const cell = hierarchicalGrid.getCellByColumn(2, 'ProductName');
+        cell.nativeElement.focus();
```
Comment (nrobakova) [inline projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.navigation.spec.ts]: In order cell to be focused you have to use cell.nativeElement.dispatchEvent(new Event('focus'));

File: projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.navigation.spec.ts [hunk 3]
```diff
@@ -1192,6 +1192,52 @@ describe('IgxHierarchicalGrid Smaller Child Navigation #hGrid', () => {
...
+        await wait(100);
+        fixture.detectChanges();
+
+        const keyboardEvent = new KeyboardEvent('keydown', {
+            key: 'Tab'
+        });
+        cell.dispatchEvent(keyboardEvent);
```
Comment (nrobakova) [inline projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.navigation.spec.ts]: There is defined function for keyboard iterations:
UIInteractions.triggerKeyDownEvtUponElem('Tab', cell.nativeElement, true);

File: projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.navigation.spec.ts [hunk 4]
```diff
@@ -1289,3 +1335,31 @@ export class IgxHierarchicalGridMultiLayoutComponent extends IgxHierarchicalGrid
...
+        <igx-row-island [key]="'childData'" [autoGenerate]="true" [expandChildren]='true' [height]="'200px'" #rowIsland>
+        </igx-row-island>
+    </igx-hierarchical-grid>`
+})
+export class IgxHierarchicalGridSpecificScenarioComponent extends IgxHierarchicalGridTestBaseComponent {
+    constructor() {
+        super();
+        this.data = this.generateData(20, 2);
+    }
+    generateData(count: number, level: number) {
```
Comment (nrobakova) [inline projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.navigation.spec.ts]: As this method is used more than once you can define it in the sample-test-data.spec.ts. An reuse it in all the definitions.

**Pull request merged**